### PR TITLE
[11.0][NEW] website_sale_product_minimal_price: New module

### PR DESCRIPTION
**Pull request by sergio-teruel**
cc @Tecnativa

### CODE REVIEW

File: website_sale_product_minimal_price/README.rst [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+**Table of contents**
+
+.. contents::
+   :local:
+
+Usage
+=====
+
+#. Go to backend and set a product with variants and extra price by attribute
```
Comment (pedrobaeza) [inline website_sale_product_minimal_price/README.rst:40]: s/and/an

File: website_sale_product_minimal_price/README.rst [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+   value.
+#. Go to Website Shop.
+#. You will see that in main products view appears the text "From " with
+   minimal price .
```
Comment (pedrobaeza) [inline website_sale_product_minimal_price/README.rst]: Remove leading space before `.`

File: website_sale_product_minimal_price/README.rst [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+.. contents::
+   :local:
+
+Usage
+=====
+
+#. Go to backend and set a product with variants and extra price by attribute
+   value.
+#. Go to Website Shop.
+#. You will see that in main products view appears the text "From " with
```
Comment (pedrobaeza) [inline website_sale_product_minimal_price/README.rst:43]: `From` text appears always or only when there are multiple prices? It should be the second IMO. Please specify this in the README when done.

File: website_sale_product_minimal_price/i18n/es.po [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+msgid "<span>From </span>"
+msgstr "<span>Desde </span>"
+
+#. module: website_sale_product_minimal_price
+#: model:product.product,name:website_sale_product_minimal_price.product_product_test_1
+#: model:product.product,name:website_sale_product_minimal_price.product_product_test_2
+#: model:product.template,name:website_sale_product_minimal_price.product_product_test_1_product_template
+#: model:product.template,name:website_sale_product_minimal_price.product_product_test_2_product_template
+msgid "My product test"
+msgstr ""
```
Comment (pedrobaeza) [inline website_sale_product_minimal_price/i18n/es.po]: Please translate all

File: website_sale_product_minimal_price/data/demo.xml [hunk 5]
```diff
@@ -0,0 +1,49 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+
+    <record id="product_attribute_test" model="product.attribute">
```
Comment (pedrobaeza) [inline website_sale_product_minimal_price/data/demo.xml:4]: I prefer to not put this as demo data if it's for tests.

File: website_sale_product_minimal_price/models/product_template.py [hunk 6]
```diff
@@ -0,0 +1,17 @@
+# Copyright 2019 Tecnativa - Sergio Teruel
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import models
+
+
+class ProductTemplate(models.Model):
+    _inherit = 'product.template'
+
+    def _website_price(self):
+        templates = self.filtered(lambda x: x.product_variant_count > 1)
```
Comment (pedrobaeza) [inline website_sale_product_minimal_price/models/product_template.py:22]: This should be for variants with different prices, not with variants only.

File: website_sale_product_minimal_price/views/templates.xml [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+        </xpath>
+        <xpath expr="//li[@t-foreach='variant_id.value_ids']" position="attributes">
+            <attribute name="t-foreach">variant_id.value_ids.sorted(key=lambda x: x.price_extra)</attribute>
+        </xpath>
+    </template>
+
+    <template id="products_item" inherit_id="website_sale.products_item">
+        <!-- Display "From" if variants have distinct extra prices -->
+        <xpath expr="//span[@t-esc='product.website_price']" position="before">
+            <t t-if="product.product_variant_count &gt; 1 and len(set(product.product_variant_ids.mapped('price_extra'))) != 1">
```
Comment (pedrobaeza) [inline website_sale_product_minimal_price/views/templates.xml]: For achieving this, maybe you need an extra computed stored field at template level for indicating if there are different prices.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
And please rebase for solving Travis

File: website_sale_product_minimal_price/README.rst (hunk 1 above)
Comment (sergio-teruel) [reply website_sale_product_minimal_price/README.rst:40]: I think that this is correct...

File: website_sale_product_minimal_price/README.rst (hunk 3 above)
Comment (sergio-teruel) [reply website_sale_product_minimal_price/README.rst:43]: Yes

File: website_sale_product_minimal_price/README.rst (hunk 1 above)
Comment (pedrobaeza) [reply website_sale_product_minimal_price/README.rst:40]: OK, now I get what you want to say!

**Comment by sergio-teruel** [comment]
@pedrobaeza chages done

File: website_sale_product_minimal_price/models/product_template.py [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+        store=True,
+        string='Has variants with price extra',
+    )
+
+    @api.depends('product_variant_ids.price_extra')
+    def has_variant_price_extra(self):
+        for template in self:
+            if (template.product_variant_count and len(set(
+                    template.product_variant_ids.mapped('price_extra'))) != 1):
+                template.has_variant_price_extra = True
```
Comment (Tardo) [inline website_sale_product_minimal_price/models/product_template.py]: This never can't be False if sometime is True?

File: website_sale_product_minimal_price/models/product_template.py [hunk 9]
```diff
@@ -0,0 +1,34 @@
+# Copyright 2019 Tecnativa - Sergio Teruel
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import api, fields, models
+
+
+class ProductTemplate(models.Model):
+    _inherit = 'product.template'
+
+    has_variant_price_extra = fields.Boolean(
+        compute='_compute_has_variant_price_extra',
```
Comment (Tardo) [inline website_sale_product_minimal_price/models/product_template.py]: Can't found "_compute_has_variant_price_extra"...

**Review by Tardo — CHANGES_REQUESTED** [review]

File: website_sale_product_minimal_price/models/product_template.py (hunk 9 above)
Comment (sergio-teruel) [reply website_sale_product_minimal_price/models/product_template.py]: Fixed

File: website_sale_product_minimal_price/models/product_template.py (hunk 8 above)
Comment (sergio-teruel) [reply website_sale_product_minimal_price/models/product_template.py]: The field is computed, so by default is False

**Comment by sergio-teruel** [comment]
Changes done!!

**Review by pedrobaeza — APPROVED** [review]

**Comment by sergio-teruel** [comment]
@pedrobaeza please do not merge!!

**Comment by pedrobaeza** [comment]
Ok

**Review by rafaelbn — APPROVED** [review]

File: website_sale_product_minimal_price/models/product_template.py [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+                if len(prices) > 1:
+                    template.has_distinct_variant_price = True
+
+    def _website_price(self):
+        templates = self.filtered(lambda x: x.product_variant_count > 1)
+        super(ProductTemplate, self - templates)._website_price()
+        for product in templates:
+            variant = product.product_variant_ids.sorted(
+                key=lambda p: p.website_price)[:1]
+            product.website_price = variant.website_price
```
Comment (Tardo) [inline website_sale_product_minimal_price/models/product_template.py:27]: Here can use .write

**Review by Tardo — CHANGES_REQUESTED** [review]

File: website_sale_product_minimal_price/models/product_template.py (hunk 10 above)
Comment (sergio-teruel) [reply website_sale_product_minimal_price/models/product_template.py:27]: This fields are computed.
https://github.com/odoo/odoo/blob/e40cbaa35f6f330fb79de49190bf764aff3554ad/addons/website_sale/models/product.py#L132

**Review by Tardo — APPROVED** [review]

**Pull request merged**